pkg/systemd/parser/unitfile.go

```diff
@@ -402,12 +402,14 @@ func (f *UnitFile) Parse(data string) error {
 		var line string
 		line, data = nextLine(data, 0)
 
-		// Handle multi-line continuations
-		// Note: This doesn't support comments in the middle of the continuation, which systemd does
-		if lineIsKeyValuePair(line) {
-			for len(data) > 0 && line[len(line)-1] == '\\' {
-				line, data = nextLine(origdata, len(line)+1)
-				nLines++
+		if !lineIsComment(line) {
```

Comment:
I don't know much about quadlet, so this may be a stupid question: why is this not:
```go
    if lineIsComment(line) {
        next iteration
    }
```
Is there metadata in comments? Special handling of some sort? It makes me super uncomfortable to see in-loop handling of comments instead of just immediate discard.

Reply:
Good point. In its current behavior, Quadlet copies the comments from the input file into the generated `.service` file. For this reason it handles such lines and does not skip them. 
Since I didn't want to change the behavior, I just made it disregard any possible line continuation character in comments

Reply:
The passing on of comments is nice, because they will then end up in the generated service file which can sometimes be shown by systemd.